pkg/pillar/cmd/zedrouter/zedrouter.go

```diff
@@ -2935,25 +2945,46 @@ func checkUplinkPortOverlap(ctx *zedrouterContext, network string, network1 stri
 	return false
 }
 
-// scan through existing AppNetworkStatus list to bring
-// up any AppNetwork struck in error state, while
-// contending for resource
-func scanAppNetworkStatusInErrorAndUpdate(ctx *zedrouterContext, key0 string) {
-	log.Infof("scanAppNetworkStatusInErrorAndUpdate()\n")
+// scan through existing AppNetworkStatus list and set a timer
+// to retry later
+func checkAppNetworkErrorAndStartTimer(ctx *zedrouterContext) {
+	log.Infof("checkAppNetworkErrorAndStartTimer()\n")
 	pub := ctx.pubAppNetworkStatus
 	items := pub.GetAll()
 	for key, st := range items {
 		status := cast.CastAppNetworkStatus(st)
 		config := lookupAppNetworkConfig(ctx, key)
 		if config == nil || !config.Activate ||
-			status.Error == "" || key == key0 {
+			status.Error == "" {
 			continue
 		}
 		// We wouldn't have even copied underlay/overlay
 		// networks into status. This is as good as starting
 		// from scratch all over. App num that would have been
 		// allocated will be used this time also, since the app UUID
 		// does not change.
+		// When hit error while creating, set a timer and wait for sometime and come back
+		log.Infof("checkAppNetworkErrorAndStartTimer: set timer\n")
+		ctx.appNetCreateTimer = time.NewTimer(60 * time.Second)
+	}
+}
+
+// scan through existing AppNetworkStatus list to bring
+// up any AppNetwork struck in error state, while
+// contending for resource
+func scanAppNetworkStatusInErrorAndUpdate(ctx *zedrouterContext) {
+	log.Infof("scanAppNetworkStatusInErrorAndUpdate()\n")
+	pub := ctx.pubAppNetworkStatus
+	items := pub.GetAll()
+	for key, st := range items {
+		status := cast.CastAppNetworkStatus(st)
+		config := lookupAppNetworkConfig(ctx, key)
+		if config == nil || !config.Activate ||
+			status.Error == "" {
+			continue
+		}
+		// called from the timer, run the AppNetworkCreate to retry
+		log.Infof("scanAppNetworkStatusInErrorAndUpdate: retry the AppNetworkCreate\n")
 		handleAppNetworkCreate(ctx, key, *config)
```

Comment:
If this fails to create, don't you need to restart the timer?

Reply:
yes, it does. this scanAppNetworkStatusInErrorAndUpdate() will call handleAppNetworkCreate(), and at the end of function in handleAppNetworkCreate() which calls checkAppNetworkErrorAndStartTimer() to check and start the timer again if any one of the ANS still has error.
this work flow is similar to the recursive calling but uses a timer instead. I have tested using instrumented code to let the ANS fail 10 timers in a row in the creation. It works fine.